### ISSUE TITLE
Unstable Training

### ISSUE THREAD

**Issue by wenjunli-0** [inclusionAI/ASearcher#29]
When I switch to the areal v0.3.3 and run the asearcher_local experiment, I got very unstable training and training cannot continue due to very high rollout time costs. 

Below are the hyperparameters I used to run the experiment and the logs. 
epochs=10, 4p1t1+4p1t1, batch_size=128, max_concurrent_rollouts=48, mem_fraction_static=0.80

<img width="2140" height="1330" alt="Image" src="https://github.com/user-attachments/assets/582d7479-dcf1-4705-9150-ac7e1848601b" />

<img width="1024" height="1198" alt="Image" src="https://github.com/user-attachments/assets/a8fd4e0e-6bc0-4ae8-b7e5-3a570d694fe3" />

<img width="1038" height="553" alt="Image" src="https://github.com/user-attachments/assets/13dac184-f7a1-4725-8250-836efbd7e627" />

**Comment by wenjunli-0** [comment]
I ran a few new asearcher-local experiments recently and they are normal in the beginning. However, around 200-250 steps, the rollout time starts to explode and the training hangs. I observe the grpo_actor/score/av, number_output_token/avg, num_search_queries/avg and etc, they all seem to be normal and do not explode. 

What will cause such a phenomenon, and how should I fix it?

<img src="https://github.com/user-attachments/assets/40fb39c2-5000-459c-9b0a-57ddd6ca8f82" width="50%" />
<img src="https://github.com/user-attachments/assets/a35d4a63-30c4-432f-bed9-5bc7399cbc79" width="50%" />

**Comment by Taewhoo** [comment]
@wenjunli-0 Hi, have you found a solution to this issue (rollout spike around steps 200-250) ?

**Comment by wenjunli-0** [comment]
> [@wenjunli-0](https://github.com/wenjunli-0) Hi, have you found a solution to this issue (rollout spike around steps 200-250) ?

I think it's because of the dynamic filtering, which filters out zero-advantage samples and thus causes the spike (e.g., needs more time to collect non-zero-advantage samples when the model can answer almost all of the questions in the training data).

**Comment by Taewhoo** [comment]
> > [@wenjunli-0](https://github.com/wenjunli-0) Hi, have you found a solution to this issue (rollout spike around steps 200-250) ?
> 
> I think it's because of the dynamic filtering, which filters out zero-advantage samples and thus causes the spike (e.g., needs more time to collect non-zero-advantage samples when the model can answer almost all of the questions in the training data).

That makes sense. It's difficult to reach the performance of [ASearcher-Local-7B](https://huggingface.co/inclusionAI/ASearcher-Local-7B) using AReal v0.3.3, and perhaps it's because I can only train up to around 220 steps. Maybe I should revert to the legacy version of AReal, as mentioned in [issue#2](https://github.com/inclusionAI/ASearcher/issues/2#issuecomment-3216208485)?

**Comment by wenjunli-0** [comment]
Yes, honestly, I also had difficulty with fully reproducing their experiments and results shown in the paper.
